scripts/barclamp_mgmt_lib.rb

```diff
@@ -421,24 +422,36 @@ def bc_install_layout_1_app(from_rpm, bc_path)
   debug "Component #{component} added to Crowbar Framework.  Review #{filelist} for files created."
 end
 
-# sync the chef parts
-def bc_install_layout_1_chef
+def run_rake_task(task_name, log)
+  crowbar_user = Etc.getpwnam("crowbar")
   rails_env = ENV["RAILS_ENV"] || "production"
-  upload_cmd = "cd #{CROWBAR_PATH} && RAILS_ENV=#{rails_env} bin/rake chef:upload:all"
-  system "su -s /bin/sh - crowbar sh -c \"#{upload_cmd}\""
+  debug("spawning bin/rake #{task_name}")
+  rake_process = spawn({"RAILS_ENV" => rails_env},
+                       "bin/rake --silent #{task_name}",
+                       :uid => crowbar_user.uid,
```

Comment:
I've investigated this. The previous code makes use of su -s /bin/sh - crowbar, which run as crowbar prompts for a password, so I'm wondering if you have confused su and sudo. I've been having trouble running barclamp_install as a non-root user, since it keeps wanting to write into /opt/dell.